### PR TITLE
Added sig-build ownership paths

### PR DESCRIPTION
**Pull request by amzn-changml**
Signed-off-by: Mike Chang <changml@amazon.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -54,4 +54,11 @@
...
 /Code/Tools/SceneAPI/ @o3de/sig-content-maintainers
-/Gems/SceneProcessing/ @o3de/sig-content-maintainers
\ No newline at end of file
+/Gems/SceneProcessing/ @o3de/sig-content-maintainers
+
+# SIG-Build areas of ownership
+/scripts/build @o3de/sig-build-reviewers @o3de/sig-build-maintainers
+/scripts/commit_validation @o3de/sig-build-maintainers
+/scripts/license_scanner @o3de/sig-build-maintainers
+/scripts/signer @o3de/sig-build-maintainers
```
Comment (lumberyard-employee-dm) [inline .github/CODEOWNERS:63]: Any areas of Code in the cmake folder that the build team wants to add to their areas of ownership?

**Review by brianherrera — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (amzn-changml) [reply .github/CODEOWNERS:63]: Yes, we discussed it earlier with the SIG to do a second pass, but I think we can at least take oversight of the installer scripts in `cmake/`

Comment (lumberyard-employee-dm) [reply .github/CODEOWNERS:63]: It should be fine to have overlapping areas of ownership for certain areas of the codebase.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -57,8 +57,10 @@
 /Gems/SceneProcessing/ @o3de/sig-content-maintainers
 
 # SIG-Build areas of ownership
+/cmake/Platform/ @o3de/sig-build-reviewers @o3de/sig-build-maintainers
+/cmake/Packaging/ @o3de/sig-build-reviewers @o3de/sig-build-maintainers
+/cmake/Packging* @o3de/sig-build-reviewers @o3de/sig-build-maintainers
```
Comment (lumberyard-employee-dm) [inline .github/CODEOWNERS:62]: This looks like a typo to me.

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (amzn-changml) [reply .github/CODEOWNERS]: Thanks for catching. Fixed.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by brianherrera — APPROVED** [review]

**Pull request merged**